src/actions/post_build.rs

```diff
@@ -224,14 +225,20 @@ fn parse_diagnostics(message: &str) -> Option<FileDiagnostic> {
             primary_message.push_str(&format!("\n\n{}", notes));
         }
 
+        // A diagnostic source is quite likely to be clippy if it contains
+        // the further information link to the rust-clippy project.
+        if primary_message.contains("rust-clippy") {
```

Comment:
minor nit: I think it'd be simpler if we would completely define it in one place like `let source = if ... { "clippy" } else { "rustc" }; `

Reply:
Because the detection logic is in the main diagnostic building scope, yet applied to all secondary diagnostics too I think this is a fairly graceful way. That you could return the main diagnostic plus the source in a tuple and avoid the `mut`, and while I generally agree with avoiding `mut` I don't think there's much of a win here.

Reply:
Ah, you’re right - I got confused by partially unfolded code in between, my bad!